MinitexPDFProtocols/MinitexPDFProtocols.swift

```diff
@@ -9,67 +9,92 @@
+  ///   - page: An optional page to navigate to when first opening.
+  ///   - bookmarks: Any additional user-saved bookmarks.
+  ///   - annotations: Any additional user-created generic annotations.
+  init?(file: URL,
+        openToPage page: MinitexPDFPage?,
+        bookmarks: [MinitexPDFPage]?,
+        annotations: [MinitexPDFAnnotation]?)
 }
 
+/// A page in a PDF document.
+@objc public protocol MinitexPDFPage: class {
+
+  /// A PDF document must have an unsigned integer represenation of its pages.
+  var pageNumber: UInt { get }
+}
 
 @objc public protocol MinitexPDFViewControllerDelegate: class {
-  @objc func userDidNavigate(page: Int)
-  @objc func saveBookmarks(pageNumbers: [UInt])
-  @objc func saveAnnotations(annotations: [MinitexPDFAnnotation])
-  @objc func willMoveToMinitexParentController(parent: UIViewController?)
+
+  /// Called when a page of the PDF document becomes the new presented page
+  /// visible on screen.
+  ///
+  /// - Parameter page: The page being presented on screen.
+  func userDidNavigate(toPage page: MinitexPDFPage)
+  /// Called when a user selects to create a bookmark for a specific page.
+  ///
+  /// - Parameter bookmark: The specific page the user has selected to save as a
+  ///   bookmark.
+  func userDidCreate(bookmark: MinitexPDFPage)
+  /// Called when a user selects to delete a bookmark for a specific page.
+  ///
+  /// - Parameter bookmark: The specific page the user has selected to delete.
+  func userDidDelete(bookmark: MinitexPDFPage)
+  /// Called when a user selects to create a generic annotation for a specific page.
+  ///
+  /// - Parameter bookmark: The specific generic annotation the user has selected to save as a
+  ///   bookmark.
+  func userDidCreate(annotation: MinitexPDFAnnotation)
+  /// Called when a user selects to delete a generic annotation for a specific page.
+  ///
+  /// - Parameter bookmark: The specific generic annotation the user has selected to delete.
+  func userDidDelete(annotation: MinitexPDFAnnotation)
 }
 
+//MARK: -
+//MARK: Factory Methods
 
-@objc public class MinitexPDFViewControllerFactory: NSObject {
-  @objc public static func createPDFViewController(dictionary: [String: Any]) -> MinitexPDFViewController? {
+@objcMembers public class MinitexPDFViewControllerFactory: NSObject {
 
-    guard let pdfViewControllerClass = NSClassFromString("PDF.PDFViewController")
-                                        as? MinitexPDFViewController.Type else {
-      return nil
-    }
+  /// The API host must create a View Controller for the PDF from a static factory method.
+  ///
+  /// - Parameters:
+  ///   - fileUrl: file directory of the PDF document
+  ///   - page: optional page to open to
+  ///   - bookmarks: any saved bookmarks by the user
+  ///   - annotations: any additional generic annotations created by the user
+  /// - Returns: An instance of MinitexPDFViewController, or nil if something failed.
+  public static func create(fileUrl: URL,
```

Comment:
The factory init is made to match the protocol init for the same view controller.